rfcs/rfc-involuntary-dissolutions.md

```diff
@@ -0,0 +1,210 @@
+
+- Start Date: 2024-04-26
+- Target Major Version:
+- Reference Issues: n/a
+- Entity Issue: [20937](https://github.com/bcgov/entity/issues/20937)
+- Implementation PR: (leave this empty)
+
+
+## Overview
+
+The involuntary dissolution process will consist of two BE scheduled jobs - involuntary dissolutions job and furnishings job.
+
+This process will be mostly automated with staff input only in the form of updating a configuration.  The configuration determines how many new businesses can be added to the dissolution process on a per day basis.
+
+The involuntary dissolutions job will manage the overall dissolution process for batches of businesses where the dissolution process is kicked off.
+
+The furnishings job will manage and track the notifications that need to go out as a part of the dissolution process.  This will involve emails, physical mail and any other required notifications made to external systems.  Functionality to reconcile BCMail+ physical mail outs and potentially track bounced emails(if required) may be handled in this job as well.
+
+These jobs will operate independently, but there will likely be some dependencies where one job may depend on some of the data generated by the other.  For example, the furnishings job will likely need to reference the active **batches** table data generated by the involuntary dissolutions job to determine which businesses to add to the furnishing tables to schedule the required notifications for new batches of business that have been initiated in the dissolution process.
+
+The legal api will also be extended to provide the following functionality:
+* Addition of warnings to GET /business response
+    * Eligible for involuntary dissolution i.e. multiple ARs not filed and other relevant conditions
+    * Business is in process of being involuntarily dissolved - warning level 1 (can be used by FE to represent legacy D1)
+    * Business is in process of being involuntarily dissolved - warning level 2 (can be used by FE to represent legacy D2)
```

Comment:
New warnings will require Filings UI work to display.

Also, any changes to allowable actions?

Reply:
Will the business state (or sub-state???) reflect eligibility/D1/D2?

Reply:
D1/D2 are not states, they are hold-overs from mainframe GDG processing. I think we can retire that now.

Reply:
There are things like this in the designs.  How did you see this working @thorwolpert ?  Would it just show a warning that the business is in the process of being dissolved?

![image](https://github.com/bcgov/entity/assets/6685223/a2f0db2d-6a57-4cae-903f-beabb498c7ab)

Reply:
> D1/D2 are not states, they are hold-overs from mainframe GDG processing. I think we can retire that now.

Have removed references to D1/D2 in RFC.

Also made updates to warnings section.